sky/packages/sky/lib/base/image_resource.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:sky' as sky;
+
+typedef void ImageListener(sky.Image image);
+
+class ImageResource {
+  ImageResource(this._futureImage) {
+    _futureImage.then((sky.Image image) {
+      _image = image;
+      _resolved = true;
+      _notifyListeners();
+    });
+  }
+
+  bool _resolved = false;
+  Future<sky.Image> _futureImage;
+  sky.Image _image;
+  final List<ImageListener> _listeners = new List<ImageListener>();
+
+  Future<sky.Image> get first => _futureImage;
```

Comment:
You can await a future that's already resolved.  You'll get called back asynchronously.